### PR TITLE
Abandoned check 

### PR DESCRIPTION
**Pull request by mario-deluna**
This PR introduces a workflow to identify abandoned repositories in Awesome-php.

It works incredibly simple by checking for 2 main indiciators.
1. **Archived Status**: Automatically considers a repository abandoned if marked as archived by a maintainer.
2. **Push Activity**: Checks if the last push to any branch of the repository occurred more than a set time ago. This PR uses a 4-year threshold, which can be adjusted as needed.
 
As this is aweomse-php, the scripts are written in PHP.

To function correctly, it requires the "GH_PA_TOKEN" repository secret, set with a personal access token. This token is needed to make GitHub API calls.

See the workflow in action at:
https://github.com/mario-deluna/awesome-php/actions/runs/6824568456/job/18560745274

Importantly, the workflow does not fail upon detecting an abandoned package. Instead, it adds a warning annotation to the README.md.

With the current rules the following repos are considered abandoned:

 - https://github.com/joshdifabio/composed last pushed at 2016-11-14 (archived: ✅ active)
 - https://github.com/slimphp/Slim-Views last pushed at 2022-02-10 (archived: ⛔️ yes)
 - https://github.com/FriendsOfPHP/Goutte last pushed at 2023-04-01 (archived: ⛔️ yes)
 - https://github.com/oscarotero/psr7-middlewares last pushed at 2018-01-27 (archived: ⛔️ yes)
 - https://github.com/nojacko/email-validator last pushed at 2018-07-21 (archived: ⛔️ yes)
 - https://github.com/fzaninotto/Streamer last pushed at 2017-01-31 (archived: ⛔️ yes)
 - https://github.com/danielstjules/pho last pushed at 2017-04-24 (archived: ✅ active)
 - https://github.com/jolicode/JoliCi last pushed at 2016-11-18 (archived: ⛔️ yes)
 - https://github.com/ircmaxell/RandomLib last pushed at 2018-09-06 (archived: ✅ active)
 - https://github.com/timoh6/TCrypto last pushed at 2016-08-10 (archived: ✅ active)
 - https://github.com/ircmaxell/password-policy last pushed at 2016-09-07 (archived: ✅ active)
 - https://github.com/jeremykendall/password-validator last pushed at 2018-04-07 (archived: ✅ active)
 - https://github.com/ircmaxell/PHP-PasswordLib last pushed at 2017-10-30 (archived: ✅ active)
 - https://github.com/sebastianbergmann/phpcpd last pushed at 2023-01-10 (archived: ⛔️ yes)
 - https://github.com/monque/PHP-Migration last pushed at 2018-07-20 (archived: ✅ active)
 - https://github.com/phacility/xhprof last pushed at 2019-05-28 (archived: ✅ active)
 - https://github.com/Wisembly/Elephant.io last pushed at 2022-08-25 (archived: ⛔️ yes)
 - https://github.com/wkhtmltopdf/wkhtmltopdf last pushed at 2022-11-22 (archived: ⛔️ yes)
 - https://github.com/box/spout last pushed at 2022-05-26 (archived: ⛔️ yes)
 - https://github.com/chanmix51/Pomm last pushed at 2017-05-21 (archived: ✅ active)
 - https://github.com/icomefromthenet/Migrations last pushed at 2016-10-06 (archived: ✅ active)
 - https://github.com/Cilex/Cilex last pushed at 2018-01-25 (archived: ✅ active)
 - https://github.com/triplepoint/php-units-of-measure last pushed at 2019-08-08 (archived: ✅ active)
 - https://github.com/ircmaxell/filterus last pushed at 2018-11-21 (archived: ✅ active)
 - https://github.com/romaricdrigon/MetaYaml last pushed at 2019-03-16 (archived: ✅ active)
 - https://github.com/serkin/Volan last pushed at 2018-06-19 (archived: ✅ active)
 - https://github.com/leedavis81/drest last pushed at 2017-05-11 (archived: ✅ active)
 - https://github.com/wsdl2phpgenerator/wsdl2phpgenerator last pushed at 2023-03-17 (archived: ⛔️ yes)
 - https://github.com/TimeToogo/Pinq last pushed at 2022-05-03 (archived: ⛔️ yes)
 - https://github.com/namshi/notificator last pushed at 2018-06-05 (archived: ✅ active)
 - https://github.com/rocketeers/rocketeer last pushed at 2020-09-05 (archived: ⛔️ yes)
 - https://github.com/square/connect-php-sdk last pushed at 2020-06-29 (archived: ⛔️ yes)
 - https://github.com/victorstanciu/dbv last pushed at 2018-03-26 (archived: ⛔️ yes)

### CODE REVIEW

File: scripts/checkAbandoned.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+define('APHP_GH_ANNOTATION_LEVEL', 'warning');
+
+// Find all github repos in our Awesome PHP list
+$awesomeListContent = getAwesomeListContents();
+$githubRepos = findUrlsInMarkdown('github.com', $awesomeListContent);
+
+// Github action annotations
+$annotations = [];
+
+foreach($githubRepos as $repoUrl) {
```
Comment (dereuromark) [inline scripts/checkAbandoned.php:22]: Can you maybe run the [code-sniffer](https://github.com/php-collective/code-sniffer) over the PHP code?

File: scripts/checkAbandoned.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                "::%s file=%s,line=%d,col=0::Abandoned repository, last push at '%s' (archived: %s)", 
+                APHP_GH_ANNOTATION_LEVEL,
+                APHP_NAME_MD, 
+                $line,
+                date('Y-m-d', $lastPush),
+                $isArchived ? 'yes' : 'no'
+            );
+        }
+        
+        printlnRed(" - {$repoData['full_name']} last pushed at " . date('Y-m-d', $lastPush) . " (archived: " . ($isArchived ? 'yes' : 'no') . ")");
```
Comment (dereuromark) [inline scripts/checkAbandoned.php:60]:
```suggestion
        printlnRed(" - {$repoData['full_name']} last pushed at " . date('Y-m-d', $lastPush) . " (status: " . ($isArchived ? 'archived' : 'active') . ")");
```

File: scripts/checkAbandoned.php (hunk 1 above)
Comment (mario-deluna) [reply scripts/checkAbandoned.php]: Done! I also ran phpstan L9 over it and fixed those.

**Review by dereuromark — APPROVED** [review]
I can add my github access token prior to merging.

**Pull request merged**